source/Calamari/Deployment/Features/NginxFeature.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Calamari.Integration.FileSystem;
+using Calamari.Integration.Nginx;
+using Calamari.Integration.Processes;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Octostache;
+
+namespace Calamari.Deployment.Features
+{
+    public class NginxFeature : IFeature
+    {
+        public string Name => "Octopus.Features.Nginx";
+        public string DeploymentStage => DeploymentStages.AfterDeploy;
+
+        public void Execute(RunningDeployment deployment)
+        {
+            var variables = deployment.Variables;
+
+            var nginxServer = NginxServer.AutoDetect();
+
+            var (rootLocation, additionalLocations) = GetLocations(variables);
+            if (rootLocation == null) throw new NginxMissingRootLocationException();
+
+            var enabledBindings = GetEnabledBindings(variables).ToList();
+            var sslCertificates = GetSslCertificates(enabledBindings, variables);
+            
+            nginxServer
+                .WithVirtualServerName(variables.Get(SpecialVariables.Package.NuGetPackageId))
+                .WithHostName(variables.Get(SpecialVariables.Action.Nginx.Server.HostName))
+                .WithServerBindings(enabledBindings, sslCertificates)
+                .WithRootLocation(rootLocation)
+                .WithAdditionalLocations(additionalLocations);
+
+            Log.Verbose("Building nginx configuration");
+            nginxServer.BuildConfiguration();
+
+            Log.Verbose("Saving nginx configuration");
+            var tempDirectory = CalamariPhysicalFileSystem.GetPhysicalFileSystem().CreateTemporaryDirectory();
```

Comment:
There is no guarantee this will be cleaned up.  An error saving the configuration will leave the temporary directory because the post deploy script never runs.